canonical-web-and-design/maas-ui
# Build SectionHeader component using common patterns from MachineList and KVMList.

## Done

- Built SectionHeader component using common patterns from MachineList and KVMList (title, subtitle, button, form wrapper, tabs) and used in machine list, KVM list and KVM details.
- Renamed `title` prop in Section to `header` which I think better implies the header can be more than just text
- Renamed machine list header components to be more specific, e.g. HeaderStrip => MachineListHeader.
- Converted machine list header components to TS.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Check that the machine list and KVM list headers still behave as before (check the Add X buttons, check that action forms still load and run properly, check that the title and subtitle display correctly)
